Client/Frontend/Browser/MainMenuActionHelper.swift

```diff
@@ -627,6 +630,7 @@ class MainMenuActionHelper: PhotonActionSheetProtocol, FeatureFlaggable, CanRemo
             self.profile.readingList.createRecordWithURL(url.absoluteString, title: tab.title ?? "", addedBy: UIDevice.current.name)
             TelemetryWrapper.recordEvent(category: .action, method: .add, object: .readingListItem, value: .pageActionMenu)
             self.delegate?.showToast(message: .AppMenu.AddToReadingListConfirmMessage, toastAction: .addToReadingList, url: nil)
+            self.notificationCenter.post(name: .ReadingListUpdated, object: self)
```

Comment:
If we go that route, on top of my head there's other places that we would need to update reading list and bookmarks section.
Ex: 
- We add a reading list item from the reader mode (on a reader mode available website, we click on the book plus icon in the reader view control)
- We manually add a bookmark for the bookmarks panel (edit > + > bookmark)

Reply:
Good catch, ideally there would be only one place where the actual data for both lists is updated. I'll dig around and see if there is anywhere like that at the moment, if not then for now I'll add multiple notifications but I'll keep track of it as something to come back to later as a candidate to potentially refactor.